CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs

```diff
@@ -0,0 +1,290 @@
+		private static (string, string) GetOptionsParameter(SnippetModel model, IndentManager indentManager, string payloadParam, string queryParamsParam, string headersParam) {
+			var nonEmptyParameters = new string[] { payloadParam, queryParamsParam, headersParam}.Where(p => !string.IsNullOrEmpty(p));
+			if(nonEmptyParameters.Any()) {
+                var className = $"msgraphsdk.{model.PathNodes.Last().GetClassName("RequestBuilder").ToFirstCharacterUpperCase()}{model.Method.ToString().ToLowerInvariant().ToFirstCharacterUpperCase()}Options";
+    			var payloadSB = new StringBuilder();
+				payloadSB.AppendLine($"{indentManager.GetIndent()}{optionsParameterVarName} := &{className}{{");
+                indentManager.Indent();
+                if(!string.IsNullOrEmpty(payloadParam))
+                    payloadSB.AppendLine($"{indentManager.GetIndent()}Body: {payloadParam},");
+                if(!string.IsNullOrEmpty(queryParamsParam))
+                    payloadSB.AppendLine($"{indentManager.GetIndent()}Q: {queryParamsParam},");
+                if(!string.IsNullOrEmpty(headersParam))
+                    payloadSB.AppendLine($"{indentManager.GetIndent()}H: {headersParam},");
+                indentManager.Unindent();
+				payloadSB.AppendLine($"{indentManager.GetIndent()}}}");
+				return (payloadSB.ToString(), optionsParameterVarName);
+            } else return (string.Empty, "nil");
+		}
+		private const string requestParametersVarName = "requestParameters";
+		private static (string, string) GetRequestQueryParameters(SnippetModel model, IndentManager indentManager) {
+			var payloadSB = new StringBuilder();
+			if(!string.IsNullOrEmpty(model.QueryString)) {
+                var className = $"msgraphsdk.{model.PathNodes.Last().GetClassName("RequestBuilder").ToFirstCharacterUpperCase()}{model.Method.ToString().ToLowerInvariant().ToFirstCharacterUpperCase()}QueryParameters";
+				payloadSB.AppendLine($"{indentManager.GetIndent()}{requestParametersVarName} := &{className}{{");
+				indentManager.Indent();
+				var (queryString, replacements) = ReplaceNestedOdataQueryParameters(model.QueryString);
+				foreach(var queryParam in queryString.TrimStart('?').Split('&', StringSplitOptions.RemoveEmptyEntries)) {
+					if(queryParam.Contains("=")) {
+						var kvPair = queryParam.Split('=', StringSplitOptions.RemoveEmptyEntries);
+						payloadSB.AppendLine($"{indentManager.GetIndent()}{NormalizeQueryParameterName(kvPair[0])}: {GetQueryParameterValue(kvPair[1], replacements)},");
+					} else
+						payloadSB.AppendLine($"{indentManager.GetIndent()}{NormalizeQueryParameterName(queryParam)}: \"\",");
+				}
+				indentManager.Unindent();
+				payloadSB.AppendLine($"{indentManager.GetIndent()}}}");
+				return (payloadSB.ToString(), requestParametersVarName);
+			}
+			return (default, default);
+		}
+		private static Regex nestedStatementRegex = new(@"(\w+)(\([^)]+\))", RegexOptions.IgnoreCase);
```

Comment:
```suggestion
		private static Regex nestedStatementRegex = new(@"(\w+)(\([^)]+\))", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```
I think we can still do this?